### PR TITLE
Refactor filenames to camelCase and remove unused files

### LINKED ISSUE #67: Refactor filenames
- All files not using the camel case naming convention should be renamed appropriately. 
- Also file with unnecessary prefix such as RatParseClassToStructs.m or RAT_parse_cells.m should be rename to remove prefix.
- files with duplicates, 'new' suffix and other similar should also be dealt with

### PR DESCRIPTION
**Pull request by DrPaulSharp**
Filenames have been changed throughout the code to change to camelCase, remove unnecessary "RAT" prefixes, and/or make the purpose of the contents clearer. This closes #67 
Some unused files have also been removed.

### CODE REVIEW

File: targetFunctions/common/loopMatlabCustomLayers/sourceCompile/loopMatlabCustomLayersWrapperXYPoints.m [hunk 1]
```diff
@@ -1,4 +1,4 @@
-function [allLayers, allRoughs] = loopMatalbCustlayWrapper_XYContrasts(cBacks,cShifts,cScales,cNbas,cNbss,cRes,backs,...
+function [allLayers, allRoughs] = loopMatlabCustomLayersWrapperXYPoints(cBacks,cShifts,cScales,cNbas,cNbss,cRes,backs,...
      shifts,sf,nba,nbs,res,cCustFiles,numberOfContrasts,customFiles,params)
```
Comment (StephenNneji) [inline targetFunctions/common/loopMatlabCustomLayers/sourceCompile/loopMatlabCustomLayersWrapperXYPoints.m:2]: `loopMatalbCustlayWrapper_XYContrasts` -> `loopMatlabCustomLayersWrapperXYPoints`?

File: tests/testCommonFunctions/testCommonFunctions.m [hunk 2]
```diff
@@ -156,16 +156,16 @@ function loadMakeSLDProfileXY(testCase)
             testCase.makeSLDProfileXYOutputs = outputs.outputs;
         end
 
-        function loadabeles_paraPoints(testCase)
-            inputs = load('abeles_paraPointsInputs.mat');
-            outputs = load('abeles_paraPointsOutputs.mat');
+        function loadabelesParallelPoints(testCase)
```
Comment (StephenNneji) [inline tests/testCommonFunctions/testCommonFunctions.m:159]: should be `loadAbelesParallelPoints`

**Review by StephenNneji — COMMENTED** [review]
examples/manual/DSPCStandardLayers/t1toProblemDef.m to .../t1ToProblemDef.m

we should remove 
targetFunctions/common/loopCppCustlayWrapper/testDLL_mex.mexa64
targetFunctions/common/loopCppCustlayWrapper/testDLL_mex_latest.mexa64

File: targetFunctions/common/loopCppCustomLayersWrapper/loopCppCustomLayersWrapperCustomLayersPoints.m [hunk 3]
```diff
@@ -1,4 +1,4 @@
-function [allLayers,allRoughs] =  loopCppCustlayWrapper_CustLaypoints(cBacks,cShifts,cScales,cNbas,cNbss,cRes,backs,...
+function [allLayers,allRoughs] =  loopCppCustomLayersWrapperCustomLayersPoints(cBacks,cShifts,cScales,cNbas,cNbss,cRes,backs,...
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply targetFunctions/common/loopCppCustomLayersWrapper/loopCppCustomLayersWrapperCustomLayersPoints.m:1]: My thoughts exactly. I did mention it in Monday's meeting as well, so let's go for it now.

File: targetFunctions/common/loopMatlabCustomLayers/sourceCompile/loopMatlabCustomLayersWrapperXYPoints.m (hunk 1 above)
Comment (DrPaulSharp) [reply targetFunctions/common/loopMatlabCustomLayers/sourceCompile/loopMatlabCustomLayersWrapperXYPoints.m:2]: So, the function name now matches the filename. Why didn't it before?

Comment (DrPaulSharp) [reply targetFunctions/common/loopMatlabCustomLayers/sourceCompile/loopMatlabCustomLayersWrapperXYPoints.m:2]: They do match on the master branch - I don't know where the contrasts comes from.

File: targetFunctions/common/loopMatlabCustomLayers/mexCompile/loopMatlabCustomLayersWrapperCustomLayersSingle.m [hunk 4]
```diff
@@ -20,7 +20,7 @@
  % We do this using feval, which automatically makes this function call
  % only extrinsic... the loop is then handled in the matlab file (using 
  % paralell computing toolbox)
- [tempAllLayers, tempAllRoughs] = feval('loopMatlabCustomLayers_points',cBacks,cShifts,cScales,cNbas,cNbss,cRes,backs,...
+ [tempAllLayers, tempAllRoughs] = feval('loopMatlabCustomLayersSingle',cBacks,cShifts,cScales,cNbas,cNbss,cRes,backs,...
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply targetFunctions/common/loopMatlabCustomLayers/mexCompile/loopMatlabCustomLayersWrapperCustomLayersSingle.m:23]: Yikes.

[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply targetFunctions/common/loopMatlabCustomLayers/mexCompile/loopMatlabCustomLayersWrapperCustomLayersSingle.m:23]: Hold on, on the master branch it is 'loopMatlabCustomLayers_single', and I've changed it to camelCase. Where does points come from?

File: compile/RAT_main_compile/RAT_main.m [hunk 5]
```diff
@@ -37,22 +37,22 @@
         outProblemDef = problemDef;
     case 'simplex'
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply compile/RAT_main_compile/RAT_main.m:38]: Okay to leave it for now

File: compile/simplex/ratSendTextOutput.m [hunk 6]
```diff
@@ -1,6 +1,6 @@
-function ratSendTextOutput(outText)
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply compile/simplex/ratSendTextOutput.m:1]: Should that file still be there? It doesn't seem to come from the build process (hence the filename is not updated).

File: targetFunctions/common/resampleLayers/SLDFunction.m [hunk 7]
```diff
@@ -16,7 +16,7 @@
 % 
 % SLD = sldProfile;
 
-%SLD = getappdata(0,'sldFuncSLD');
+%SLD = getappdata(0,'SLDFunctionSLD');
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply targetFunctions/common/resampleLayers/SLDFunction.m:19]: This line is commented out, and it's the only mention of SLD re. "appdata" in the entire code.

File: compile/simplex/ratSendTextOutput.m (hunk 6 above)
[the remark this answers is not in the document]
Comment (StephenNneji) [reply compile/simplex/ratSendTextOutput.m:1]: Yes its the mex build for macOS. Arwel said rebuilding it every time was a pain for some reason so I left it in the repo but will remove after the macOS build server is up. It should be renamed to match the mex call

File: targetFunctions/common/resampleLayers/SLDFunction.m (hunk 7 above)
[the remark this answers is not in the document]
Comment (StephenNneji) [reply targetFunctions/common/resampleLayers/SLDFunction.m:19]: Sorry I missed that it was a comment

[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply targetFunctions/common/resampleLayers/SLDFunction.m:19]: Fair

File: minimisers/DREAM/functions/RATEvaluateModel.m [hunk 8]
```diff
@@ -1,4 +1,4 @@
-function fx = ratEvaluate_model(x,DREAMPar,Meas_info,ratInputs)
+function fx = RATEvaluateModel(x,DREAMPar,Meas_info,ratInputs)
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/RATEvaluateModel.m:1]: `evaluateRATModel` better?

File: minimisers/DREAM/functions/Whittle.m [hunk 9]
```diff
@@ -1,4 +1,4 @@
-function [log_L] = Whittle(Meas_info,fx);
+function [log_L] = Whittle(Meas_info,fx)
 % Calculates whittle's log-likelihood function using spectral analysis
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/Whittle.m:2]: should be `whittle`

File: minimisers/DREAM/postprocessing/genparset.m [hunk 10]
```diff
@@ -1,4 +1,4 @@
-function ParSet = GenParSet(chain);
+function ParSet = genparset(chain)
```
Comment (StephenNneji) [inline minimisers/DREAM/postprocessing/genparset.m:1]: should be `genParSet`

File: minimisers/DREAM/functions/RATDREAMInitialize.m [hunk 11]
```diff
@@ -1,4 +1,4 @@
-function [chain,output,X,fx,CR,pCR,lCR,delta_tot,log_L] = ratDREAM_initialize(DREAMPar,Par_info,Meas_info,chain,output,log_L,ratInputs)
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/RATDREAMInitialize.m:1]: `initializaDREAMForRAT` better?

File: minimisers/DREAM/functions/RATDREAMCalcSetup.m [hunk 12]
```diff
@@ -1,4 +1,4 @@
-function [DREAMPar] = ratDREAM_calc_setup(DREAMPar)
+function [DREAMPar] = RATDREAMCalcSetup(DREAMPar)
 % Sets up sequential / parallel
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/RATDREAMCalcSetup.m:2]: `setupDREAMCalc` maybe?

File: minimisers/DREAM/functions/MeasError.m [hunk 13]
```diff
@@ -1,4 +1,4 @@
-function [out] = MeasError(Y);
+function [out] = MeasError(Y)
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/MeasError.m:1]: should be `measError`

File: minimisers/DREAM/functions/Latin.m [hunk 14]
```diff
@@ -1,4 +1,4 @@
-function x = LHS(minn,maxn,N)
+function x = Latin(minn,maxn,N)
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/Latin.m:1]: should be `latinHypercubeSampling`. latin is to generic

File: minimisers/DREAM/functions/GenParSet.m [hunk 15]
```diff
@@ -1,4 +1,4 @@
-function ParSet = GenParSet(chain);
+function ParSet = GenParSet(chain)
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/GenParSet.m:1]: There is an identical GenParSet in postprocessing folder. I think we can delete one

File: minimisers/DREAM/functions/Gelman.m [hunk 16]
```diff
@@ -12,7 +12,7 @@
 % Compute the dimensions of chain
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/Gelman.m:12]: should be `gelman`

File: minimisers/DREAM/functions/DREAMStoreResults.m [hunk 17]
```diff
@@ -1,4 +1,4 @@
-function DREAM_store_results ( DREAMPar , fx , Meas_info , id );
+function DREAMStoreResults ( DREAMPar , fx , Meas_info , id )
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/DREAMStoreResults.m:1]: `storeDREAMResults` better?

File: minimisers/DREAM/functions/DREAMSetup.m [hunk 18]
```diff
@@ -1,5 +1,5 @@
 function [outDREAMPar,Par_info,Meas_info,chain,output,log_L,Table_gamma,iloc,iteration,...
-        gen] = DREAM_setup(DREAMPar,Par_info,Meas_info)
+        gen] = DREAMSetup(DREAMPar,Par_info,Meas_info)
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/DREAMSetup.m:2]: `setupDream` better?

File: minimisers/DREAM/functions/DREAMInitialize.m [hunk 19]
```diff
@@ -1,4 +1,4 @@
-function [chain,output,X,fx,CR,pCR,lCR,delta_tot,log_L] = DREAM_initialize(DREAMPar,Par_info,Meas_info,chain,output,log_L,ratInputs)
+function [chain,output,X,fx,CR,pCR,lCR,delta_tot,log_L] = DREAMInitialize(DREAMPar,Par_info,Meas_info,chain,output,log_L,ratInputs)
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/DREAMInitialize.m:1]: `initializaDREAM `better?

File: minimisers/DREAM/functions/DREAMCheck.m [hunk 20]
```diff
@@ -1,4 +1,4 @@
-function stop = DREAM_check(DREAMPar,Par_info,Meas_info);
+function stop = DREAMCheck(DREAMPar,Par_info,Meas_info)
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/DREAMCheck.m:1]: `checkDREAM`

File: minimisers/DREAM/functions/DREAMCalcSetup.m [hunk 21]
```diff
@@ -1,4 +1,4 @@
-function [DREAMPar] = DREAM_calc_setup(DREAMPar)
+function [DREAMPar] = DREAMCalcSetup(DREAMPar)
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/DREAMCalcSetup.m:1]: `setDREAMParam`

File: minimisers/DREAM/functions/DREAMCalcSetupORIG.m [hunk 22]
```diff
@@ -1,4 +1,4 @@
-function [DREAMPar,f_handle] = DREAM_calc_setup(DREAMPar,Func_name)
```
Comment (StephenNneji) [inline minimisers/DREAM/functions/DREAMCalcSetupORIG.m:1]: Please remove this file if its not used

File: minimisers/DREAM/functions/RATDREAMCalcSetup.m (hunk 12 above)
Comment (DrPaulSharp) [reply minimisers/DREAM/functions/RATDREAMCalcSetup.m:2]: I think `setupDREAMForRAT` given `setupDREAM` already exists

**Pull request merged**